### PR TITLE
bochs: Fix build on Big Sur

### PR DESCRIPTION
**Pull request by y-rath**
#### Description

Fix macOS version checking in configure script to pick up the right flags.

See: [#61488](https://trac.macports.org/ticket/61488)

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 11.0.1 20B29
Xcode 12.2 12B45b

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by ryandesign** [comment]
This is not the same as the patch we are recommending people use (from the libtool port) and it doesn't look correct.

`case ${MACOSX_DEPLOYMENT_TARGET}` would never contain a comma, so it would never match `10.[012],*|,*powerpc*)`.

**Comment by y-rath** [comment]
@ryandesign Yeah, you're right. Careless on my part (perils of copy-paste). Can you please review the updated fix?

**Comment by ryandesign** [comment]
Your new fix is still different from the one we used in the libtool port but should be ok. Now we just have to wait for the maintainer @Nax to approve the PR.

**Review by Nax — APPROVED** [review]

**Pull request merged**
